controllers/deactivation/deactivation_controller.go

```diff
@@ -139,9 +139,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	// If the deactivation timeout is 0 then users that belong to this tier should not be automatically deactivated
 	if deactivationTimeoutDays == 0 {
 		// If the usersignup was already set to deactivating then reset it to false
-		if err := r.resetDeactivatingState(ctx, usersignup); err != nil {
+		updated, err := r.resetDeactivatingState(ctx, usersignup)
+		if err != nil {
 			return reconcile.Result{}, err
 		}
+
+		// If the UserSignup was updated, reload it
+		if updated {
```

Comment:
Maybe something like this instead?
```
usersignup, err := r.resetDeactivatingState(ctx, usersignup)

func (r *Reconciler) resetDeactivatingState(ctx context.Context, usersignup *toolchainv1alpha1.UserSignup) (*toolchainv1alpha1.UserSignup, error) {
    us := usersignup
    ...
    if updated {
        us = ... // Get the updated UserSignup
    }
    return us, nil
}

```

Reply:
Actually why not just simply exit the reconcile loop if we updated the status? The controller watches the entire usersignup including the status. So when we change it then it will trigger another reconcile loop.

Reply:
I actually thought about doing it this way after, I think it will make for neater code.

Reply:
BTW I don't believe it's necessary to reload the UserSignup after all updates, only when it's possible for multiple updates to occur in the same reconcile.